mobile/src/test/java/org/openhab/habdroid/model/OpenHAB2WidgetTest.java

```diff
@@ -1,67 +1,404 @@
-        OpenHABWidget sut = OpenHAB2Widget.createOpenHABWidgetFromJson(rootWidget, createJSONObject(), "PNG");
-        assertTrue(sut instanceof OpenHAB2Widget);
-    }
-
-    private JSONObject createJSONObject() throws Exception {
-
-        String json = "{\n" +
-                "    \"widgetId\": \"demo\",\n" +
-                "    \"type\": \"Group\",\n" +
-                "    \"label\": \"Group1\",\n" +
-                "    \"icon\": \"groupicon\",\n" +
-                "    \"url\": \"http://localhost/url\",\n" +
-                "    \"minValue\": \"0.0\",\n" +
-                "    \"maxValue\": \"10.0\",\n" +
-                "    \"step\": \"1\",\n" +
-                "    \"refresh\": \"10\",\n" +
-                "    \"period\": \"D\",\n" +
-                "    \"service\": \"D\",\n" +
-                "    \"height\": \"10\",\n" +
-                "    \"iconcolor\": \"white\",\n" +
-                "    \"labelcolor\": \"white\",\n" +
-                "    \"valuecolor\": \"white\",\n" +
-                "    \"mappings\": [{\n" +
-                "      \"command\": \"ON\",\n" +
-                "      \"label\": \"On\"\n" +
-                "    }],\n" +
-                "    \"item\": {\n" +
-                "      \"type\": \"GroupItem\",\n" +
-                "      \"name\": \"group1\",\n" +
-                "      \"state\": \"OFF\",\n" +
-                "      \"link\": \"http://localhost/rest/items/group1\"\n" +
-                "    },\n" +
-                "    \"linkedPage\": {\n" +
-                "      \"id\": \"0001\",\n" +
-                "      \"title\": \"LinkedPage\",\n" +
-                "      \"icon\": \"linkedpageicon\",\n" +
-                "      \"link\": \"http://localhost/rest/sitemaps/demo/0001\",\n" +
-                "      \"leaf\": \"false\"\n" +
-                "    },\n" +
-                "    \"widgets\": [{ \"widgetId\": \"demo11\" }]\n" +
-                "  }";
+    public void testGetIconPath() throws Exception {
+        assertEquals("icon/groupicon?state=OFF&format=PNG", sut1.getIconPath());
+        assertEquals("icon/groupicon?state=ON&format=SVG", sut2.getIconPath());
+        assertEquals("icon/slider?state=81&format=SVG", sut3.getChildren().get(0).getIconPath());
+
+        // Rollersutter icon should always be 0 to 100, not ON/OFF
+        //assertEquals("icon/rollershutter?state=0&format=SVG", sut3.getChildren().get(1).getIconPath());
+        assertEquals("icon/rollershutter?state=42&format=SVG", sut3.getChildren().get(2).getIconPath());
+
+    }
+
+    @Test
+    public void testGetChildren() throws Exception {
+        assertEquals("demo11", sut1.getChildren().get(0).getId());
+
+        // sut2 doesn't have children
+        IndexOutOfBoundsException e = null;
+        try {
+            sut2.getChildren().get(0).getId();
+        } catch (IndexOutOfBoundsException ex) {
+            e = ex;
+        }
+        assertTrue(e != null);
```

Comment:
assertNotNull
However, what do you think about having this as a separate test (testGetChildrenNoChildren or whatever) and use the test annotations expected:
@Test(expected = IndexOutOfBoundsException.class)